#Contents List
##Import Libraries
##Create Path
##Import Data - orders_products_merged_updated.pkl
##2. In this Exercise, you learned how to find the aggregated mean of the “order_number” column grouped by “department_id” for a subset of your dataframe. Now, repeat this process for the entire dataframe.
##3. Analyze the result. How do the results for the entire dataframe differ from those of the subset? Include your comments in a markdown cell below the executed code.
##4. Follow the instructions in the Exercise for creating a loyalty flag for existing customers using the transform() and loc() functions.
##5. The marketing team at Instacart wants to know whether there’s a difference between the spending habits of the three types of customers you identified. Use the loyalty flag you created and check the basic statistics of the product prices for each loyalty category (Loyal Customer, Regular Customer, and New Customer). What you’re trying to determine is whether the prices of products purchased by loyal customers differ from those purchased by regular or new customers.
##6. The team now wants to target different types of spenders in their marketing campaigns. This can be achieved by looking at the prices of the items people are buying. Create a spending flag for each user based on the average price across all their orders using the following criteria:
##Marking values above $25.00 as missing - Nan
##Aggregating the data
##Transform the data
##Create a spending flag
##Value Counts
###7. In order to send relevant notifications to users within the app (for instance, asking users if they want to buy the same item again), the Instacart team wants you to determine frequent versus non-frequent customers. Create an order frequency flag that marks the regularity of a user’s ordering behavior according to the median in the “days_since_prior_order” column. The criteria for the flag should be as follows:
##Aggregating the data
##Transform the data
##Create a frequency flag
##Value Counts
##9. Export your dataframe as a pickle file and store it correctly in your “Prepared Data” folder. - orders_products_merged_updated_2.pkl

In [1]:
#Import Libraries
import pandas as pd
import numpy as np
import os

In [2]:
#Path
path = r'C:\Users\Family Computer\Documents\Data Analytics\Instacart Basket Analysis'

In [3]:
#Import ords_prods_merge_updated dataframe
df_ords_prods_merge_updated = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'orders_products_merged_updated.pkl'))

In [4]:
#2. In this Exercise, you learned how to find the aggregated mean of the “order_number” column grouped by “department_id” for a subset of your dataframe. Now, repeat this process for the entire dataframe.
df_ords_prods_merge_updated.groupby('department_id').agg({'order_number':['mean']})

order_number
                      mean
department_id             
1                15.457838
2                17.277920
3                17.170395
4                17.811403
5                15.215751
6                16.439806
7                17.225802
8                15.340650
9                15.895474
10               20.197148
11               16.170638
12               15.887671
13               16.583536
14               16.773669
15               16.165037
16               17.665606
17               15.694469
18               19.310397
19               17.177343
20               16.473447
21               22.902379

#3. Analyze the result. How do the results for the entire dataframe differ from those of the subset? Include your comments in a markdown cell below the executed code.
##The subset only included results from 8 of the 21 departments. We can now see the actual averages of all 21 departments, whereas when we limited the dataframe to 1,000,000 rows, the averages were also based on those 1,000,000 rows. For example, department 4 had an average of 18.825780 when we only looked at 1,000,000 rows. Now that we are looking at the entire dataframe, we can see that the actual average for department 4 is 17.811403. 

In [5]:
#4. Follow the instructions in the Exercise for creating a loyalty flag for existing customers using the transform() and loc() functions.
df_ords_prods_merge_updated['max_order']=df_ords_prods_merge_updated.groupby(['user_id'])['order_number'].transform(np.max)

C:\Users\Family Computer\AppData\Local\Temp\ipykernel_26648\3681961856.py:2: FutureWarning: The provided callable <function amax at 0x0000029CE9D00220> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  df_ords_prods_merge_updated['max_order']=df_ords_prods_merge_updated.groupby(['user_id'])['order_number'].transform(np.max)


In [6]:
df_ords_prods_merge_updated.head(15)

order_id  user_id  order_number  orders_day_of_week  hour_of_day_ordered  \
0    2539329        1             1                   2                    8   
1    2398795        1             2                   3                    7   
2     473747        1             3                   3                   12   
3    2254736        1             4                   4                    7   
4     431534        1             5                   4                   15   
5    3367565        1             6                   2                    7   
6     550135        1             7                   1                    9   
7    3108588        1             8                   1                   14   
8    2295261        1             9                   1                   16   
9    2550362        1            10                   4                    8   
10   2968173       15            15                   1                    9   
11   1870022       15            17                   2                   16   
12   1911383       15            18                   2                   11   
13   2715276       15            21                   1                    9   
14    487368       15            22                   1                   10   

    days_since_prior_order  product_id  add_to_cart_order  reordered  \
0                      NaN         196                  1          0   
1                     15.0         196                  1          1   
2                     21.0         196                  1          1   
3                     29.0         196                  1          1   
4                     28.0         196                  1          1   
5                     19.0         196                  1          1   
6                     20.0         196                  1          1   
7                     14.0         196                  2          1   
8                      0.0         196                  4          1   
9                     30.0         196                  1          1   
10                     7.0         196                  2          0   
11                     8.0         196                  6          1   
12                     7.0         196                  1          1   
13                     7.0         196                  1          1   
14                    14.0         196                  1          1   

   product_name  aisle_id  department_id  prices _merge    price_range_loc  \
0          Soda        77              7     9.0   both  Mid-range product   
1          Soda        77              7     9.0   both  Mid-range product   
2          Soda        77              7     9.0   both  Mid-range product   
3          Soda        77              7     9.0   both  Mid-range product   
4          Soda        77              7     9.0   both  Mid-range product   
5          Soda        77              7     9.0   both  Mid-range product   
6          Soda        77              7     9.0   both  Mid-range product   
7          Soda        77              7     9.0   both  Mid-range product   
8          Soda        77              7     9.0   both  Mid-range product   
9          Soda        77              7     9.0   both  Mid-range product   
10         Soda        77              7     9.0   both  Mid-range product   
11         Soda        77              7     9.0   both  Mid-range product   
12         Soda        77              7     9.0   both  Mid-range product   
13         Soda        77              7     9.0   both  Mid-range product   
14         Soda        77              7     9.0   both  Mid-range product   

     busiest day busiest_period_of_day  max_order  
0   Regular busy        Average orders         10  
1   Slowest days        Average orders         10  
2   Slowest days           Most orders         10  
3   Slowest days        Average orders         10  
4   Slowest days           Most orders         10  
5   Regul

In [7]:
df_ords_prods_merge_updated.loc[df_ords_prods_merge_updated['max_order'] >40, 'loyalty_flag'] = 'Loyal customer'

In [8]:
df_ords_prods_merge_updated.loc[(df_ords_prods_merge_updated['max_order'] <=40) & (df_ords_prods_merge_updated['max_order'] >=10), 'loyalty_flag'] = 'Regular customer'

In [9]:
df_ords_prods_merge_updated.loc[df_ords_prods_merge_updated['max_order'] <=10, 'loyalty_flag'] = 'New customer'

In [10]:
df_ords_prods_merge_updated['loyalty_flag'].value_counts (dropna = False)

loyalty_flag
Regular customer    15876776
Loyal customer      10284093
New customer         6243990
Name: count, dtype: int64

In [11]:
df_ords_prods_merge_updated[['user_id', 'loyalty_flag', 'order_number']].head(60)

user_id      loyalty_flag  order_number
0         1      New customer             1
1         1      New customer             2
2         1      New customer             3
3         1      New customer             4
4         1      New customer             5
5         1      New customer             6
6         1      New customer             7
7         1      New customer             8
8         1      New customer             9
9         1      New customer            10
10       15  Regular customer            15
11       15  Regular customer            17
12       15  Regular customer            18
13       15  Regular customer            21
14       15  Regular customer            22
15       19      New customer             2
16       19      New customer             5
17       19      New customer             7
18       21  Regular customer            10
19       31  Regular customer            10
20       31  Regular customer            17
21       43  Regular customer             6
22       43  Regular customer             9
23       52  Regular customer             1
24       52  Regular customer             2
25       52  Regular customer             3
26       52  Regular customer             5
27       52  Regular customer             7
28       52  Regular customer             9
29       52  Regular customer            10
30       52  Regular customer            11
31       52  Regular customer            12
32       52  Regular customer            15
33       52  Regular customer            17
34       52  Regular customer            18
35       52  Regular customer            19
36       52  Regular customer            21
37       67  Regular customer             1
38       67  Regular customer             2
39       67  Regular customer             3
40       67  Regular customer             4
41       67  Regular customer             5
42       67  Regular customer             6
43       67  Regular customer             8
44       67  Regular customer             9
45       67  Regular customer            10
46       67  Regular customer            11
47       67  Regular customer            12
48       67  Regular customer            13
49       67  Regular customer            14
50       67  Regular customer            16
51       67  Regular customer            18
52       67  Regular customer            19
53       67  Regular customer            21
54       67  Regular customer            22
55       67  Regular customer            23
56       81      New customer             3
57       81      New customer             5
58       82  Regular customer             1
59       82  Regular customer             3

In [12]:
#5. The marketing team at Instacart wants to know whether there’s a difference between the spending habits of the three types of customers you identified. Use the loyalty flag you created and check the basic statistics of the product prices for each loyalty category (Loyal Customer, Regular Customer, and New Customer). What you’re trying to determine is whether the prices of products purchased by loyal customers differ from those purchased by regular or new customers.
df_ords_prods_merge_updated.groupby('loyalty_flag').agg({'prices':['mean', 'median', 'min', 'max']})

prices                     
                       mean median  min      max
loyalty_flag                                    
Loyal customer    10.386336    7.4  1.0  99999.0
New customer      13.294670    7.4  1.0  99999.0
Regular customer  12.495717    7.4  1.0  99999.0

In [13]:
#6. The team now wants to target different types of spenders in their marketing campaigns. This can be achieved by looking at the prices of the items people are buying. Create a spending flag for each user based on the average price across all their orders using the following criteria:
## If the mean of the prices of products purchased by a user is lower than 10, then flag them as a “Low spender.”
## If the mean of the prices of products purchased by a user is higher than or equal to 10, then flag them as a “High spender.”
df_ords_prods_merge_updated.groupby('product_name')

In [14]:
#Marking values above $25.00 as missing - Nan
df_ords_prods_merge_updated.loc[df_ords_prods_merge_updated['prices'] >100, 'prices']=np.nan

In [15]:
df_ords_prods_merge_updated['prices'].max()

25.0

In [16]:
### Aggregating the data
df_ords_prods_merge_updated.groupby('product_name').agg({'prices':['mean']})

prices
                                                     mean
product_name                                             
#2 Coffee Filters                                    11.0
#2 Cone White Coffee Filters                          5.7
#2 Mechanical Pencils                                 9.2
#4 Natural Brown Coffee Filters                       2.5
#NAME?                                                3.6
...                                                   ...
with Xylitol Unwrapped Original Flavor 50 Stick...    5.4
with Xylitol Unwrapped Spearmint 50 Sticks Suga...    5.2
with Xylitol Watermelon Twist 18 Sticks Sugar F...    3.5
with a Splash of Mango Coconut Water                  2.6
with a Splash of Pineapple Coconut Water              4.2

[49661 rows x 1 columns]

In [17]:
### Transform the data
df_ords_prods_merge_updated['average_price']=df_ords_prods_merge_updated.groupby(['user_id'])['prices'].transform (np.mean)

C:\Users\Family Computer\AppData\Local\Temp\ipykernel_26648\4036302361.py:2: FutureWarning: The provided callable <function mean at 0x0000029CE9D00F40> is currently using SeriesGroupBy.mean. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "mean" instead.
  df_ords_prods_merge_updated['average_price']=df_ords_prods_merge_updated.groupby(['user_id'])['prices'].transform (np.mean)


In [18]:
df_ords_prods_merge_updated.head(15)

order_id  user_id  order_number  orders_day_of_week  hour_of_day_ordered  \
0    2539329        1             1                   2                    8   
1    2398795        1             2                   3                    7   
2     473747        1             3                   3                   12   
3    2254736        1             4                   4                    7   
4     431534        1             5                   4                   15   
5    3367565        1             6                   2                    7   
6     550135        1             7                   1                    9   
7    3108588        1             8                   1                   14   
8    2295261        1             9                   1                   16   
9    2550362        1            10                   4                    8   
10   2968173       15            15                   1                    9   
11   1870022       15            17                   2                   16   
12   1911383       15            18                   2                   11   
13   2715276       15            21                   1                    9   
14    487368       15            22                   1                   10   

    days_since_prior_order  product_id  add_to_cart_order  reordered  \
0                      NaN         196                  1          0   
1                     15.0         196                  1          1   
2                     21.0         196                  1          1   
3                     29.0         196                  1          1   
4                     28.0         196                  1          1   
5                     19.0         196                  1          1   
6                     20.0         196                  1          1   
7                     14.0         196                  2          1   
8                      0.0         196                  4          1   
9                     30.0         196                  1          1   
10                     7.0         196                  2          0   
11                     8.0         196                  6          1   
12                     7.0         196                  1          1   
13                     7.0         196                  1          1   
14                    14.0         196                  1          1   

   product_name  aisle_id  department_id  prices _merge    price_range_loc  \
0          Soda        77              7     9.0   both  Mid-range product   
1          Soda        77              7     9.0   both  Mid-range product   
2          Soda        77              7     9.0   both  Mid-range product   
3          Soda        77              7     9.0   both  Mid-range product   
4          Soda        77              7     9.0   both  Mid-range product   
5          Soda        77              7     9.0   both  Mid-range product   
6          Soda        77              7     9.0   both  Mid-range product   
7          Soda        77              7     9.0   both  Mid-range product   
8          Soda        77              7     9.0   both  Mid-range product   
9          Soda        77              7     9.0   both  Mid-range product   
10         Soda        77              7     9.0   both  Mid-range product   
11         Soda        77              7     9.0   both  Mid-range product   
12         Soda        77              7     9.0   both  Mid-range product   
13         Soda        77              7     9.0   both  Mid-range product   
14         Soda        77              7     9.0   both  Mid-range product   

     busiest day busiest_period_of_day  max_order      loyalty_flag  \
0   Regular busy        Average orders         10      New customer   
1   Slowest days        Average orders         10      New customer   
2   Slowest days           Most orders         10      New customer   
3   Slowest days        Average order

In [19]:
### Create a spending flag
df_ords_prods_merge_updated.loc[df_ords_prods_merge_updated['average_price'] <10, 'spending_flag']= 'Low spender'

In [20]:
df_ords_prods_merge_updated.loc[df_ords_prods_merge_updated['average_price'] >=10, 'spending_flag']= 'High spender'

In [21]:
### Value Counts
df_ords_prods_merge_updated['spending_flag'].value_counts (dropna = False)

spending_flag
Low spender     32285131
High spender      119728
Name: count, dtype: int64

In [22]:
df_ords_prods_merge_updated[['user_id', 'average_price', 'spending_flag']].head(60)

user_id  average_price spending_flag
0         1       6.367797   Low spender
1         1       6.367797   Low spender
2         1       6.367797   Low spender
3         1       6.367797   Low spender
4         1       6.367797   Low spender
5         1       6.367797   Low spender
6         1       6.367797   Low spender
7         1       6.367797   Low spender
8         1       6.367797   Low spender
9         1       6.367797   Low spender
10       15       3.980556   Low spender
11       15       3.980556   Low spender
12       15       3.980556   Low spender
13       15       3.980556   Low spender
14       15       3.980556   Low spender
15       19       8.372059   Low spender
16       19       8.372059   Low spender
17       19       8.372059   Low spender
18       21       7.012683   Low spender
19       31       8.137793   Low spender
20       31       8.137793   Low spender
21       43       7.444898   Low spender
22       43       7.444898   Low spender
23       52       6.356287   Low spender
24       52       6.356287   Low spender
25       52       6.356287   Low spender
26       52       6.356287   Low spender
27       52       6.356287   Low spender
28       52       6.356287   Low spender
29       52       6.356287   Low spender
30       52       6.356287   Low spender
31       52       6.356287   Low spender
32       52       6.356287   Low spender
33       52       6.356287   Low spender
34       52       6.356287   Low spender
35       52       6.356287   Low spender
36       52       6.356287   Low spender
37       67       7.656790   Low spender
38       67       7.656790   Low spender
39       67       7.656790   Low spender
40       67       7.656790   Low spender
41       67       7.656790   Low spender
42       67       7.656790   Low spender
43       67       7.656790   Low spender
44       67       7.656790   Low spender
45       67       7.656790   Low spender
46       67       7.656790   Low spender
47       67       7.656790   Low spender
48       67       7.656790   Low spender
49       67       7.656790   Low spender
50       67       7.656790   Low spender
51       67       7.656790   Low spender
52       67       7.656790   Low spender
53       67       7.656790   Low spender
54       67       7.656790   Low spender
55       67       7.656790   Low spender
56       81       8.165385   Low spender
57       81       8.165385   Low spender
58       82       6.441221   Low spender
59       82       6.441221   Low spender

In [23]:
#7. In order to send relevant notifications to users within the app (for instance, asking users if they want to buy the same item again), the Instacart team wants you to determine frequent versus non-frequent customers. Create an order frequency flag that marks the regularity of a user’s ordering behavior according to the median in the “days_since_prior_order” column. The criteria for the flag should be as follows:
## If the median of “days_since_prior_order” is higher than 20, then the customer should be labeled a “Non-frequent customer.”
## If the median is higher than 10 and lower than or equal to 20, then the customer should be labeled a “Regular customer.”
## If the median is lower than or equal to 10, then the customer should be labeled a “Frequent customer.”
df_ords_prods_merge_updated.groupby('user_id')

In [24]:
### Aggregating the data
df_ords_prods_merge_updated.groupby('user_id').agg({'days_since_prior_order':['median']})

days_since_prior_order
                        median
user_id                       
1                         20.5
2                         13.0
3                         10.0
4                         20.0
5                         11.0
...                        ...
206205                    30.0
206206                     3.0
206207                    16.0
206208                     7.0
206209                    22.0

[206209 rows x 1 columns]

In [25]:
### Transform the data
df_ords_prods_merge_updated['median_days_since prior_order']=df_ords_prods_merge_updated.groupby(['user_id'])['days_since_prior_order'].transform (np.median)

C:\Users\Family Computer\AppData\Local\Temp\ipykernel_26648\157217329.py:2: FutureWarning: The provided callable <function median at 0x0000029CE9E3E660> is currently using SeriesGroupBy.median. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "median" instead.
  df_ords_prods_merge_updated['median_days_since prior_order']=df_ords_prods_merge_updated.groupby(['user_id'])['days_since_prior_order'].transform (np.median)


In [26]:
df_ords_prods_merge_updated.head(15)

order_id  user_id  order_number  orders_day_of_week  hour_of_day_ordered  \
0    2539329        1             1                   2                    8   
1    2398795        1             2                   3                    7   
2     473747        1             3                   3                   12   
3    2254736        1             4                   4                    7   
4     431534        1             5                   4                   15   
5    3367565        1             6                   2                    7   
6     550135        1             7                   1                    9   
7    3108588        1             8                   1                   14   
8    2295261        1             9                   1                   16   
9    2550362        1            10                   4                    8   
10   2968173       15            15                   1                    9   
11   1870022       15            17                   2                   16   
12   1911383       15            18                   2                   11   
13   2715276       15            21                   1                    9   
14    487368       15            22                   1                   10   

    days_since_prior_order  product_id  add_to_cart_order  reordered  \
0                      NaN         196                  1          0   
1                     15.0         196                  1          1   
2                     21.0         196                  1          1   
3                     29.0         196                  1          1   
4                     28.0         196                  1          1   
5                     19.0         196                  1          1   
6                     20.0         196                  1          1   
7                     14.0         196                  2          1   
8                      0.0         196                  4          1   
9                     30.0         196                  1          1   
10                     7.0         196                  2          0   
11                     8.0         196                  6          1   
12                     7.0         196                  1          1   
13                     7.0         196                  1          1   
14                    14.0         196                  1          1   

   product_name  ...  prices  _merge    price_range_loc   busiest day  \
0          Soda  ...     9.0    both  Mid-range product  Regular busy   
1          Soda  ...     9.0    both  Mid-range product  Slowest days   
2          Soda  ...     9.0    both  Mid-range product  Slowest days   
3          Soda  ...     9.0    both  Mid-range product  Slowest days   
4          Soda  ...     9.0    both  Mid-range product  Slowest days   
5          Soda  ...     9.0    both  Mid-range product  Regular busy   
6          Soda  ...     9.0    both  Mid-range product  Busiest days   
7          Soda  ...     9.0    both  Mid-range product  Busiest days   
8          Soda  ...     9.0    both  Mid-range product  Busiest days   
9          Soda  ...     9.0    both  Mid-range product  Slowest days   
10         Soda  ...     9.0    both  Mid-range product  Busiest days   
11         Soda  ...     9.0    both  Mid-range product  Regular busy   
12         Soda  ...     9.0    both  Mid-range product  Regular busy   
13         Soda  ...     9.0    both  Mid-range product  Busiest days   
14         Soda  ...     9.0    both  Mid-range product  Busiest days   

   busiest_period_of_day max_order      loyalty_flag  average_price  \
0         Average orders        10      New customer       6.367797   
1         Average orders        10      New customer       6.367797   
2            Most orders        10      New customer       6.367797   
3         Average orders        10      New customer       6.367797   
4            Most orders        10      New cu

In [27]:
### Create a frequency flag
df_ords_prods_merge_updated.loc[df_ords_prods_merge_updated['median_days_since prior_order'] >20, 'frequency_flag']= 'Non-frequent customer'

In [28]:
df_ords_prods_merge_updated.loc[(df_ords_prods_merge_updated['median_days_since prior_order'] <=20) & (df_ords_prods_merge_updated['median_days_since prior_order'] >10), 'frequency_flag']= 'Regular customer'

In [29]:
df_ords_prods_merge_updated.loc[df_ords_prods_merge_updated['median_days_since prior_order'] <=10, 'frequency_flag']= 'Frequent customer'

In [30]:
### Value Counts
df_ords_prods_merge_updated['frequency_flag'].value_counts (dropna = False)

frequency_flag
Frequent customer        21559853
Regular customer          7208564
Non-frequent customer     3636437
nan                             5
Name: count, dtype: int64

In [31]:
df_ords_prods_merge_updated[['user_id', 'median_days_since prior_order', 'frequency_flag']].head(60)

user_id  median_days_since prior_order         frequency_flag
0         1                           20.5  Non-frequent customer
1         1                           20.5  Non-frequent customer
2         1                           20.5  Non-frequent customer
3         1                           20.5  Non-frequent customer
4         1                           20.5  Non-frequent customer
5         1                           20.5  Non-frequent customer
6         1                           20.5  Non-frequent customer
7         1                           20.5  Non-frequent customer
8         1                           20.5  Non-frequent customer
9         1                           20.5  Non-frequent customer
10       15                           10.0      Frequent customer
11       15                           10.0      Frequent customer
12       15                           10.0      Frequent customer
13       15                           10.0      Frequent customer
14       15                           10.0      Frequent customer
15       19                            6.0      Frequent customer
16       19                            6.0      Frequent customer
17       19                            6.0      Frequent customer
18       21                            7.5      Frequent customer
19       31                            6.0      Frequent customer
20       31                            6.0      Frequent customer
21       43                           10.0      Frequent customer
22       43                           10.0      Frequent customer
23       52                            8.0      Frequent customer
24       52                            8.0      Frequent customer
25       52                            8.0      Frequent customer
26       52                            8.0      Frequent customer
27       52                            8.0      Frequent customer
28       52                            8.0      Frequent customer
29       52                            8.0      Frequent customer
30       52                            8.0      Frequent customer
31       52                            8.0      Frequent customer
32       52                            8.0      Frequent customer
33       52                            8.0      Frequent customer
34       52                            8.0      Frequent customer
35       52                            8.0      Frequent customer
36       52                            8.0      Frequent customer
37       67                            7.0      Frequent customer
38       67                            7.0      Frequent customer
39       67                            7.0      Frequent customer
40       67                            7.0      Frequent customer
41       67                            7.0      Frequent customer
42       67                            7.0      Frequent customer
43       67                            7.0      Frequent customer
44       67                            7.0      Frequent customer
45       67                            7.0      Frequent customer
46       67                            7.0      Frequent customer
47       67                            7.0      Frequent customer
48       67                            7.0      Frequent customer
49       67                            7.0      Frequent customer
50       67                            7.0      Frequent customer
51       67                            7.0      Frequent customer
52       67                            7.0      Frequent customer
53       67                            7.0      Frequent customer
54       67                            7.0      Frequent customer
55       67                            7.0      Frequent customer
56       81                           30.0  Non-frequent customer
57       81                           30.0  Non-frequent customer
58       82                           11.5       Regular customer
59       82                           11.5  

In [32]:
#9. Export your dataframe as a pickle file and store it correctly in your “Prepared Data” folder.
df_ords_prods_merge_updated.to_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'orders_products_merged_updated_2.pkl'))